docs/utils.md

```diff
@@ -71,9 +73,19 @@ RemoveTrailingSlash removes trailing slash, if any
 ```go
 func TouchCopy(writer io.Writer, reader io.Reader) error
 ```
-TouchCopy is a wrapper around [io.Copy](https://godoc.org/io#Copy) which ensures that even empty source files
+Deprecated: Use TouchCopyBuffer Instead. TouchCopy is a wrapper around [io.Copy](https://godoc.org/io#Copy) which ensures that even empty source files
 (reader) will get written as an empty file. It guarantees a Write() call on the target file.
 
+#### func  TouchCopyBuffered
+
+```go
+func TouchCopyBuffered(writer io.Writer, reader io.Reader, bufferSize int)
+```
+TouchCopy is a wrapper around [io.CopyBuffer](https://godoc.org/io#CopyBuffer) which ensures that even empty source files
+(reader) will get written as an empty file. It guarantees a Write() call on the target file. Furthermore, it will use a
+buffer that is equal to bufferSize measured in bytes. If bufferSize is 0, then a default buffer of TouchCopyMinBufferSize 
```

Comment:
should we maybe mention what the min buffer size is here instead of having to reference it somewhere else?

Reply:
I am partial to keeping this as is just because if TouchCopyMinBufferSize is changed later then we would have to update  a number here to keep the docs current. I don't feel too bad about this because TouchCopyMinBufferSize is in the same pkg/file